src/service-api.ts

```diff
@@ -183,6 +228,40 @@ const plugin: FastifyPluginAsync<GraaspImportZipPluginOptions> = async (fastify,
       return items;
     },
   );
+
+  // download item as zip
+  fastify.get<{ Params: { itemId: string } }>(
+    '/zip-export/:itemId',
+    { schema: zipExport },
+    async ({ member, params: { itemId }, log }, reply) => {
+      const getItemTask = iTM.createGetTask(member, itemId);
+      const item = await runner.runSingle(getItemTask);
+
+      const archive = archiver.create('zip', { store: true });
+
+      archive.on('warning', function (err) {
+        if (err.code === 'ENOENT') {
+          log.debug(err);
+        } else {
+          throw err;
+        }
+      });
+      archive.on('error', function (err) {
+        throw err;
+      });
+
+      // pipe archive data to the response
+      reply.raw.setHeader('Content-Type', 'application/octet-stream');
+      reply.raw.setHeader('Content-Disposition', `filename="${item.name}.zip"`);
+      archive.pipe(reply.raw);
+
+      const rootPath = path.dirname(HOME_PATH);
+      await addFileToZip(item, rootPath, archive, member);
```

Comment:
`addItemToZip`? Since not everything is a file (even though it ends up being a file).